packages/agent_cli/src/main.rs

```diff
@@ -1,618 +1,699 @@
+                .subcommand(
+                    Command::new("login-url")
+                        .about("Generates a link to allow user to login")
+                )
+        )
+        .subcommand(
+            Command::new("claim")
+                .subcommand_required(true)
+                .arg(arg!(--name <TUNNEL_NAME> "name of the agent").required(false))
+                .about("Setting up a new playit agent")
+                .long_about("Provides a URL that can be visited to claim the agent and generate a secret key")
+                .subcommand(
+                    Command::new("generate")
+                        .about("Generates a random claim code")
+                )
+                .subcommand(
+                    Command::new("url")
+                        .about("Print a claim URL given the code and options")
+                        .arg(arg!(<CLAIM_CODE> "claim code"))
+                        .arg(arg!(--name [NAME] "name for the agent").default_value("from-cli"))
+                        .arg(arg!(--type [TYPE] "the agent type").default_value("self-managed"))
+                )
+                .subcommand(
+                    Command::new("exchange")
+                        .about("Exchanges the claim for the secret key")
+                        .arg(arg!(<CLAIM_CODE> "claim code (see \"claim generate\")"))
+                        .arg(arg!(--wait <WAIT_SEC> "number of seconds to wait 0=infinite").default_value("0"))
+                )
+        )
+        .subcommand(
+            Command::new("tunnels")
+                .subcommand_required(true)
+                .about("Manage tunnels")
+                .subcommand(
+                    Command::new("prepare")
+                        .about("Create a tunnel if it doesn't exist with the parameters")
+                        .arg(arg!(--type [TUNNEL_TYPE] "the tunnel type"))
+                        .arg(arg!(--name [NAME] "name of the tunnel"))
+                        .arg(arg!(<PORT_TYPE> "either \"tcp\", \"udp\", or \"both\""))
+                        .arg(arg!(<PORT_COUNT> "number of ports in a series to allocate").default_value("1"))
+                        .arg(arg!(--exact))
+                        .arg(arg!(--ignore_name))
+                )
+                .subcommand(
+                    Command::new("list")
+                        .about("List tunnels (format \"[tunnel-id] [port-type] [port-count] [public-address]\")")
+                )
+        )
+        .subcommand(
+            Command::new("run")
+                .about("Run the playit agent")
+                .arg(arg!([MAPPING_OVERRIDE] "(format \"<tunnel-id>=[<local-ip>:]<local-port> [, ..]\")").required(false).value_delimiter(','))
+        )
+        .subcommand(
+            Command::new("reset")
+                .about("removes the secret key on your system so the playit agent can be re-claimed")
+        )
+       //ipcheck subcommand (chroma)
+       .subcommand(

```

Comment:
I like the idea of having these debug commands, let's plan for the future to have more and add this under a nested sub command.

Something like `playit debug ipcheck` rather than `playit ipcheck`.